### ISSUE TITLE
Unable to determine a suitable edition, running yo v4.0.0

### ISSUE THREAD

**Issue by rgnns** [yeoman/yo#662]
## Type of issue

Bug

----

## My environment

* *OS version/details*: `macOS Catalina 10.15.7 Intel`
* *Node version:* `v16.0.0`
* *npm version:* `7.10.0`
* *Version of yo :* `4.0.0`

## Expected behavior

Calling `yo *` does something other than error.

## Current behavior

Error thrown, very long log... ends in:

```json
orphanStack: 'Error: editions-autoloader-none-broadened: Unable to determine a suitable edition, even
 after broadening.\n' +
    '    at new Errlop (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/errlop/ed
ition-es5/index.js:61:18)\n' +
    '    at Object.errtion (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/editi
ons/edition-es5/util.js:23:14)\n' +
    '    at determineEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/edi
tions/edition-es5/index.js:317:21)\n' +
    '    at solicitEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/editi
ons/edition-es5/index.js:350:16)\n' +
    '    at Object.requirePackage (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_module
s/editions/edition-es5/index.js:364:9)\n' +                                                                '    at Object.<anonymous> (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/i
stextorbinary/index.cjs:4:38)\n' +                                                                         '    at Module._compile (node:internal/modules/cjs/loader:1108:14)\n' +
    '    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)\n' +
    '    at Module.load (node:internal/modules/cjs/loader:988:32)\n' +
    '    at Function.Module._load (node:internal/modules/cjs/loader:828:14)',
  code: 'editions-autoloader-none-broadened'
```

## Steps to reproduce the behavior

1. Install `node` via `nvm`. (i.e. `nvim install node; nvm use node`, will install specified node version).
1. Install `yo` globally (i.e. `npm i -g yo generator-code`).
1. Run `yo` or any other command (i.e. `yo doctor`, `yo code`, `yo version`...)

## Command line output

Too long to paste without making it unreadable, it repeats something like the one pasted below, and ends with the error pasted above.

```
orphanStack: 'Error: editions-autoloader-edition-incompatible: The edition [TypeScript source cod
e with Import for modules] is not compatible with this environment.\n' +
        '    at new Errlop (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/errlo
p/edition-es5/index.js:61:18)\n' +
        '    at Object.errtion (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/e
ditions/edition-es5/util.js:23:14)\n' +
        '    at isCompatibleEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modu
les/editions/edition-es5/index.js:252:19)\n' +
        '    at determineEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules
/editions/edition-es5/index.js:287:4)\n' +
        '    at determineEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules
/editions/edition-es5/index.js:312:12)\n' +
        '    at solicitEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/e
ditions/edition-es5/index.js:350:16)\n' +
        '    at Object.requirePackage (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_mo
dules/editions/edition-es5/index.js:364:9)\n' +
        '    at Object.<anonymous> (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modul
es/istextorbinary/index.cjs:4:38)\n' +
        '    at Module._compile (node:internal/modules/cjs/loader:1108:14)\n' +
        '    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1137:10)',
      code: 'editions-autoloader-edition-incompatible'
    },
    Error: editions-autoloader-invalid-engines: The edition had no engines to compare against the envir
onment
        at new Errlop (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/errlop/edi
tion-es5/index.js:61:18)                                                                                       at Object.errtion (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/editio
ns/edition-es5/util.js:23:14)                                                                                  at isCompatibleEngines (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/e
ditions/edition-es5/index.js:183:19)
        at isCompatibleEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/e
ditions/edition-es5/index.js:250:10)
        at determineEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/edit
ions/edition-es5/index.js:287:4)
        at determineEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/edit
ions/edition-es5/index.js:312:12)
        at solicitEdition (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/editio
ns/edition-es5/index.js:350:16)
        at Object.requirePackage (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules
/editions/edition-es5/index.js:364:9)
        at Object.<anonymous> (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/is
textorbinary/index.cjs:4:38)
        at Module._compile (node:internal/modules/cjs/loader:1108:14) {
      klass: [Function: Errlop] {
        create: [Function (anonymous)],
        isErrlop: [Function (anonymous)],
        ensure: [Function (anonymous)]
      },
      parent: undefined,
      ancestors: [],
      orphanStack: 'Error: editions-autoloader-invalid-engines: The edition had no engines to compare a
gainst the environment\n' +
        '    at new Errlop (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/errlo
p/edition-es5/index.js:61:18)\n' +
        '    at Object.errtion (/Users/gl/.nvm/versions/node/v16.0.0/lib/node_modules/yo/node_modules/e
ditions/edition-es5/util.js:23:14)\n' +
```

**Comment by mshima** [comment]
Seems to be a bug at istextorbinary package. They need to add support to node 16.
Please open a bug report at https://github.com/bevry/istextorbinary/issues.

**Comment by tathagata** [comment]
Downgraded to 15.14.0

```
nvm install 15.14.0 --reinstall-packages-from=16.0.0
```

**Comment by sdsalyer** [comment]
> 
> 
> Downgraded to 15.14.0
> 
> ```
> nvm install 15.14.0 --reinstall-packages-from=16.0.0
> ```

Thank you for this. Don't forget to `nvm use 15.14.0` afterwards, and then `npm install -g yo`.

**Comment by scalvert** [comment]
Potential fix: https://github.com/bevry/istextorbinary/pull/201

**Comment by mshima** [comment]
Should be fixed in 4.1.0
